### PR TITLE
Invoke promise fix

### LINKED ISSUE #141: Cancel invokation events if already transitioned
Hi!

Is there a way to stop an invokation from sending the `done` event if we already transitioned away from the state where it started?

The `done` is a fixed event name there can be multiple states listening for that event, which means that a `done` event for a past state can trigger a transition for the "wrong" state.

Example: https://svelte.dev/repl/074d23edf65a4a4a82c68e2f5ec16b1a?version=3.12.1

This will toggle the state between `active` / `inactive` every two seconds via the `done` event from a promise.
A click on the button  will send a `toggle` event to transition between the same states.

Click the button rapidly a few times to see the strange behavior.

The intended behavior here is to toggle between the states efter 2 seconds _unless_ the user clicks the button, then it should transition directly and then restart the 2 second timer (and transition again unless user interrupts with a click).

How should I model this to get my intended behavior?

### PR DESCRIPTION
**Pull request by ehuelsmann**
Closes #171 by resolving the conflicts preventing auto-merge.
Fixes #141.

### CODE REVIEW

**Comment by ehuelsmann** [comment]
@matthewp, I'm totally lost on changesets. Could you please instruct me what the changeset bot is expecting from me? Do I need to create a *new* changeset? Or can this PR be added to an existing changeset?

**Comment by matthewp** [comment]
@ehuelsmann you create a changeset with `npm run changeset` and follow the instructions. It's just a way to automate the release process.

**Comment by ehuelsmann** [comment]
@matthewp , thanks! I've managed to add a changeset now. Is this PR good to go now? Or is there anything I can do to expedite it?

File: packages/core/test/test-invoke.js [hunk 1]
```diff
@@ -52,6 +52,50 @@ QUnit.module('Invoke', hooks => {
     assert.equal(service.machine.current, 'two', 'in the new state');
   });
 
+  QUnit.test('Should not fire "done" event when state chnages', async assert => {
```
Comment (edwardsnjd) [inline packages/core/test/test-invoke.js:55]:
```suggestion
  QUnit.test('Should not fire "done" event when state changes', async assert => {
```

File: packages/core/test/test-invoke.js [hunk 2]
```diff
@@ -52,6 +52,50 @@ QUnit.module('Invoke', hooks => {
...
+    });
+
+    let service = interpret(machine, () => { });
+    service.send('click');
+    service.send('click');
+    await wait(15)()
+    assert.equal(service.machine.current, 'three', 'now in the next state');
+  });
+
+  QUnit.test('Should fire "done" when context chnages', async assert => {
```
Comment (edwardsnjd) [inline packages/core/test/test-invoke.js:77]:
```suggestion
  QUnit.test('Should fire "done" when context changes', async assert => {
```

**Review by edwardsnjd — COMMENTED** [review]
Thanks for following up on this point of friction, I'd also love to see this land. I just spotted some typos.

**Comment by ehuelsmann** [comment]
@edwardsnjd thanks for the typo fixes and the heads up. @matthewp , anythinv we can do further to expedite this PR?

**Comment by ehuelsmann** [comment]
@matthewp Can this be merged?

Also, I sent you a mail about your progress with @gkiely as a contributor. I'm still ready to become a contributor (and merge this PR myself?). If that's stalling on his response, please don't hesitate to expedite this PR regardless.

**Pull request merged**